pkg/network/vmi/status/updater.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Authors.
+ *
+ */
+
+package status
```

Comment:
I'm a bit uncomfortable with this new package. We do not really need a VMI representative package, not a `status` one when we have `vmispec` which manipulates the spec (and status).

IMO it fits the `vmispec` package, but it seems that the dependency on the `namescheme` will cause a dependency loop.
Can you check if we can somehow solve this?

Reply:
How about moving the relevant pieces of `pkg/network/vmispec` under `pkg/network/vmi/spec` and the rest to `pkg/network/vmi/status`?

Nevertheless I will check try to see if it fits under `pkg/network/vmispec`.

Reply:
Another option is to create a new package under `vmispec`, but I do not have a good name for it. The current file name `update` is a verb, which usually does not fit.

Maybe we should create a VMI controller (`vmicontroller`) package and in it this `status` file.
We could then centralize all VMI hookpoints operations through it, at least the general ones.
WDYT?

Reply:
Another option is to create a new package under `vmispec`, but I do not have a good name for it. The current file name `update` is a verb, which usually does not fit.

Maybe we should create a VMI controller (`vmicontroller`) package and in it this `status` file.
We could then centralize all VMI hookpoints operations through it, at least the general ones.
WDYT?

Reply:
Removed the `pkg/network/vmi` package.
Created a `pkg/network/vmicontroller` package and moved the code there as suggested.